neuraxle/data_container.py

```diff
@@ -142,42 +146,95 @@ def hash_summary(self):
+                                or absent value null object to trim the absent values from the batch
+        :param include_incomplete_pass:
         :rtype: Iterable[DataContainer]
 
         .. seealso::
             `<https://github.com/guillaume-chevalier/python-conv-lib>`_
         """
-        conv_current_ids = convolved_1d(stride=stride, iterable=self.current_ids, kernel_size=kernel_size,
-                                        include_incomplete_pass=True)
-        conv_data_inputs = convolved_1d(stride=stride, iterable=self.data_inputs, kernel_size=kernel_size,
-                                        include_incomplete_pass=True)
-        conv_expected_outputs = convolved_1d(stride=stride, iterable=self.expected_outputs, kernel_size=kernel_size,
-                                             include_incomplete_pass=True)
+        if default_value_expected_outputs is None:
+            default_value_expected_outputs = default_value
+
+        conv_current_ids = convolved_1d(
+            stride=stride,
+            iterable=self.current_ids,
+            kernel_size=kernel_size,
+            include_incomplete_pass=include_incomplete_pass
+        )
+
+        conv_data_inputs = convolved_1d(
+            stride=stride,
+            iterable=self.data_inputs,
+            kernel_size=kernel_size,
+            default_value=default_value,
+            include_incomplete_pass=include_incomplete_pass
+        )
+
+        conv_expected_outputs = convolved_1d(
+            stride=stride,
+            iterable=self.expected_outputs,
+            kernel_size=kernel_size,
+            default_value=default_value_expected_outputs,
+            include_incomplete_pass=include_incomplete_pass
+        )
 
-        for current_ids, data_inputs, expected_outputs in zip(conv_current_ids, conv_data_inputs,
-                                                              conv_expected_outputs):
+        for current_ids, data_inputs, expected_outputs in zip(
+            conv_current_ids,
+            conv_data_inputs,
+            conv_expected_outputs
+        ):
             for i, (ci, di, eo) in enumerate(zip(current_ids, data_inputs, expected_outputs)):
                 if di is None:
                     current_ids = current_ids[:i]
                     data_inputs = data_inputs[:i]
                     expected_outputs = expected_outputs[:i]
                     break
 
-            yield DataContainer(
+            data_container = DataContainer(
                 data_inputs=data_inputs,
                 current_ids=current_ids,
                 summary_id=self.summary_id,
                 expected_outputs=expected_outputs,
                 sub_data_containers=self.sub_data_containers
             )
 
+            if isinstance(default_value, AbsentValuesNullObject):
+                data_container.trim(from_current_id=None)
+
+            yield data_container
+
+    def trim(self, from_current_id: Any) -> 'DataContainer':
```

Comment:
`from_current_id` should be renamed as it seems like it wants to set a "to" instead and that we trim starting from 0 and not from something else than 0. Also review type of this param.